### PR TITLE
Document System.String 2.X APIs

### PR DESCRIPTION
**Pull request by carlossanlop**
I made several white space alterations so make sure to review "ignoring white space changes".

Only one API left to review in this file: `<Member MemberName="Create&lt;TState&gt;">`. The code is [here](https://github.com/dotnet/coreclr/blob/4aa65f077778c7c05211825f98684aae9c52caaf/src/System.Private.CoreLib/shared/System/String.cs#L328). I couldn't find a PR where it was initially introduced (seems to be really old). It was only moved around during the existence of corefx/coreclr/corert. I will document it later, unless someone would like to help me figure out what it's for and I can add it in this PR.

### CODE REVIEW

**Comment by carlossanlop** [comment]
> Only one API left to review in this file: <Member MemberName="Create&lt;TState&gt;">. The code is here. I couldn't find a PR where it was initially introduced (seems to be really old). It was only moved around during the existence of corefx/coreclr/corert. I will document it later, unless someone would like to help me figure out what it's for and I can add it in this PR.

@GrabYourPitchforks @tarekgh would you mind helping adding a summary for the `Create<TState>` method? It's the only one left to document for the whole `String` class!

File: xml/System/String.xml [hunk 1]
```diff
@@ -4466,10 +4466,18 @@ Depending on Why you want to call the `Copy` method, there are a number of alter
         <Parameter Name="value" Type="System.Char" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1" />
       </Parameters>
       <Docs>
-        <param name="value">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="value">The character to compare to the character at the end of this instance.</param>
+        <summary>Determines whether the end of this string instance matches the specified character.</summary>
```
Comment (tarekgh) [inline xml/System/String.xml]: >acte [](start = 94, length = 4)

would be good if you can specify the "value" parameter.

File: xml/System/String.xml [hunk 2]
```diff
@@ -4466,10 +4466,18 @@ Depending on Why you want to call the `Copy` method, there are a number of alter
...
+        <returns><see langword="true" /> if <paramref name="value" /> matches the end of this instance; otherwise, <see langword="false" />.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+This method performs a word (case-sensitive and culture-sensitive) comparison using the current culture.
```
Comment (tarekgh) [inline xml/System/String.xml]: This method performs case-sensitive and culture-sensitive comparison using the current culture.

File: xml/System/String.xml [hunk 3]
```diff
@@ -8063,11 +8070,17 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="separator">The character to use as a separator. <paramref name="separator" /> is included in the returned string only if <paramref name="value" /> has more than one element.</param>
+        <param name="values">An array of objects from which their string representations will be concatenated.</param>
+        <summary>Concatenates the string representations of an array of objects, using the specified separator between each member.</summary>
+        <returns>A string that consists of the elements of <paramref name="values" /> delimited by the <paramref name="separator" /> character.
+
+-or-
+
+<see cref="F:System.String.Empty" /> if <paramref name="values" /> has no elements or <paramref name="separator" /> and all the elements of <paramref name="values" /> are <see cref="F:System.String.Empty" />.</returns>
```
Comment (tarekgh) [inline xml/System/String.xml]: >or <paramref name="separator" /> and all the elements of <paramref name="values" /> are <see cref="F:System.String.Empty" [](start = 83, length = 121)

This doesn't look correct. 

doing String.Join(",", new object[] {"", "", "", "" }) returns string like ",,," and not empty string

Comment (tarekgh) [reply xml/System/String.xml]: ignore this. I missed the part that the separator is empty too.

---
In reply to: [328368295](https://github.com/dotnet/dotnet-api-docs/pull/3238#discussion_r328368295) [](ancestors = 328368295)

**Comment by tarekgh** [comment]
@carlossanlop 

> would you mind helping adding a summary for the Create<TState> method? It's the only one left to document for the whole String class!

```
Summary: 
Creates a new string with specific length and allow initialize the string after creating it through the action callback.
```

Feel free to rephrase it.

File: xml/System/String.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System/String.xml]: > Creates a new string with specific length and allow initialize the string after creating it through the action callback.
> Feel free to rephrase it.

Thank you, @tarekgh.

I actually found the PR where this method was merged (not sure why I couldn't find it before):
https://github.com/dotnet/coreclr/pull/13849/files
And here is the PR with the unit tests, so I could understand how it worked:
https://github.com/dotnet/corefx/pull/23872/files

**Comment by carlossanlop** [comment]
> Creates a new string with specific length and allow initialize the string after creating it through the action callback.
> Feel free to rephrase it.

Thank you, @tarekgh.

I actually found the PR where this method was merged (not sure why I couldn't find it before):
https://github.com/dotnet/coreclr/pull/13849/files
And here is the PR with the unit tests, so I could understand how it worked:
https://github.com/dotnet/corefx/pull/23872/files

File: xml/System/String.xml (hunk 1 above)
Comment (carlossanlop) [reply xml/System/String.xml]: Do you mean like this?:
```suggestion
        <summary>Determines whether the end of this string instance matches the specified <paramref name="value" /> character.</summary>
```

File: xml/System/String.xml (hunk 2 above)
Comment (carlossanlop) [inline xml/System/String.xml]:
```suggestion
This method performs a case-sensitive and culture-sensitive comparison using the current culture.
```

File: xml/System/String.xml [hunk 4]
```diff
@@ -13888,31 +14061,31 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
- ]]></format>
+>  If the <xref:System.String.Trim()> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading and trailing white space characters found in the current instance are removed.
+
+If the current string equals <xref:System.String.Empty> or all the characters in the current instance consist of white-space characters, the method returns <xref:System.String.Empty>.
+
+White-space characters are defined by the Unicode standard. The <xref:System.String.Trim> method removes any leading and trailing characters that produce a return value of `true` when they are passed to the <xref:System.Char.IsWhiteSpace%2A?displayProperty=nameWithType> method.
+
+## Examples
+
+The following example uses the <xref:System.String.Trim()?displayProperty=nameWithType> method to remove any extra white space from strings entered by the user before concatenating them.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The following example uses the `Trim` method to remove any extra white space from strings entered by the user before concatenating them.
```

File: xml/System/String.xml [hunk 5]
```diff
@@ -14204,9 +14411,20 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Removes all the leading white-space characters from the current string.</summary>
+        <returns>The string that remains after all white-space characters are removed from the start of the current string. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimStart()> method removes from the current string all leading white-space characters. The trim operation stops when a non white-space character is encountered. For example, if the current string is "   abc   xyz    ", the <xref:System.String.TrimStart()> method returns "abc   xyz    ".
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The `TrimStart` method removes from the current string all leading white-space characters. The trim operation stops when a non white-space character is encountered. For example, if the current string is "   abc   xyz    ", the `TrimStart` method returns "abc   xyz    ".
```

File: xml/System/String.xml [hunk 6]
```diff
@@ -14003,31 +14190,31 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
- ]]></format>
+>  If the <xref:System.String.Trim(System.Char[])> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading and trailing `trimChars` characters found in the current instance are removed.
+
+If the current string equals <xref:System.String.Empty> or all the characters in the current instance consist of characters in the `trimChars` array, the method returns <xref:System.String.Empty>.
+
+If `trimChars` is `null` or an empty array, this method removes any leading or trailing characters that result in the method returning `true` when they are passed to the <xref:System.Char.IsWhiteSpace%2A?displayProperty=nameWithType> method.
+
+## Examples
+
+The following example uses the <xref:System.String.Trim(System.Char[])?displayProperty=nameWithType> method to remove space, asterisk (*), and apostrophe (') characters from a string.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The following example uses the `Trim(System.Char[])` method to remove space, asterisk (*), and apostrophe (') characters from a string.
```

File: xml/System/String.xml [hunk 7]
```diff
@@ -14063,9 +14250,18 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
       </ReturnValue>
       <Parameters />
       <Docs>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <summary>Removes all the trailing white-space characters from the current string.</summary>
+        <returns>The string that remains after all white-space characters are removed from the end of the current string. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
        <returns>The string that remains after all white-space characters are removed from the end of the current string. If no characters can be trimmed from the current instance, the method returns the current instance unchanged.</returns>
```

File: xml/System/String.xml [hunk 8]
```diff
@@ -14063,9 +14250,18 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+        <format type="text/markdown"><![CDATA[
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
              <remarks>  
                <format type="text/markdown"><![CDATA[
```

File: xml/System/String.xml [hunk 9]
```diff
@@ -14063,9 +14250,18 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
+## Remarks
+
+The <xref:System.String.TrimEnd()> method removes from the current string all trailing white-space characters. The trim operation stops when the first non white-space character is encountered at the end of the string. For example, if the current string is "   abc   xyz    ", the <xref:System.String.TrimEnd()> method returns "   abc   xyz".
+
+> [!NOTE]
+>  If the <xref:System.String.TrimEnd()> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing white-space characters are removed from the current string.
+
+          ]]></format>
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
          ]]></format>
          </remarks>
```

File: xml/System/String.xml [hunk 10]
```diff
@@ -14063,9 +14250,18 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+        <summary>Removes all the trailing white-space characters from the current string.</summary>
+        <returns>The string that remains after all white-space characters are removed from the end of the current string. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
+        <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimEnd()> method removes from the current string all trailing white-space characters. The trim operation stops when the first non white-space character is encountered at the end of the string. For example, if the current string is "   abc   xyz    ", the <xref:System.String.TrimEnd()> method returns "   abc   xyz".
+
+> [!NOTE]
+>  If the <xref:System.String.TrimEnd()> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing white-space characters are removed from the current string.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
>  If the `TrimEnd` method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing white-space characters are removed from the current string.
```

File: xml/System/String.xml [hunk 11]
```diff
@@ -14095,10 +14291,21 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
         <Parameter Name="trimChar" Type="System.Char" Index="0" FrameworkAlternate="netcore-2.0;netcore-2.1;netcore-2.2;netcore-3.0;netstandard-2.1" />
       </Parameters>
       <Docs>
-        <param name="trimChar">To be added.</param>
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
-        <remarks>To be added.</remarks>
+        <param name="trimChar">An Unicode character to remove.</param>
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
        <param name="trimChar">A Unicode character to remove.</param>
```

File: xml/System/String.xml [hunk 12]
```diff
@@ -14095,10 +14291,21 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+        <summary>Removes all the trailing occurrences of a character from the current string.</summary>
+        <returns>The string that remains after all occurrences of the <paramref name="trimChar" /> character are removed from the end of the current string. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimEnd(System.Char)> method removes from the current string all trailing `trimChar` characters. The trim operation stops when the first character that is not `trimChar` is encountered at the end of the string. For example, if `trimChar` is `-` and the current string is "---abc---xyz----", the <xref:System.String.TrimEnd(System.Char)> method returns "---abc---xyz".
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The `TrimEnd(System.Char)` method removes from the current string all trailing `trimChar` characters. The trim operation stops when the first character that is not `trimChar` is encountered at the end of the string. For example, if `trimChar` is `-` and the current string is "---abc---xyz----", the <xref:System.String.TrimEnd(System.Char)> method returns "---abc---xyz".
```

File: xml/System/String.xml [hunk 13]
```diff
@@ -14095,10 +14291,21 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
+> [!NOTE]
+>  If the <xref:System.String.TrimEnd(System.Char)> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing `trimChar` characters are removed from the current string.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
>  If the `TrimEnd(System.Char)` method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing `trimChar` characters are removed from the current string.
```

File: xml/System/String.xml [hunk 14]
```diff
@@ -14149,26 +14356,26 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-        <summary>Removes all trailing occurrences of a set of characters specified in an array from the current <see cref="T:System.String" /> object.</summary>
-        <returns>The string that remains after all occurrences of the characters in the <paramref name="trimChars" /> parameter are removed from the end of the current string. If <paramref name="trimChars" /> is <see langword="null" /> or an empty array, Unicode white-space characters are removed instead. If no characters can be trimmed from the current instance, the method returns the current instance unchanged.</returns>
+        <summary>Removes all the trailing occurrences of a set of characters specified in an array from the current string.</summary>
+        <returns>The string that remains after all occurrences of the characters in the <paramref name="trimChars" /> parameter are removed from the end of the current string. If <paramref name="trimChars" /> is <see langword="null" /> or an empty array, Unicode white-space characters are removed instead. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
        <returns>The string that remains after all occurrences of the characters in the <paramref name="trimChars" /> parameter are removed from the end of the current string. If <paramref name="trimChars" /> is <see langword="null" /> or an empty array, Unicode white-space characters are removed instead. If no characters can be trimmed from the current instance, the method returns the current instance unchanged.</returns>
```

File: xml/System/String.xml [hunk 15]
```diff
@@ -14149,26 +14356,26 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-          <format type="text/markdown"><![CDATA[  
-  
-## Remarks  
- The <xref:System.String.TrimEnd%2A> method removes from the current string all trailing characters that are in the `trimChars` parameter. The trim operation stops when the first character that is not in `trimChars` is encountered at the end of the string. For example, if the current string is "123abc456xyz789" and `trimChars` contains the digits from "1" through "9", the <xref:System.String.TrimEnd%2A> method returns "123abc456xyz".  
-  
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimEnd(System.Char[])> method removes from the current string all trailing characters that are in the `trimChars` parameter. The trim operation stops when the first character that is not in `trimChars` is encountered at the end of the string. For example, if the current string is "123abc456xyz789" and `trimChars` contains the digits from "1" through "9", the <xref:System.String.TrimEnd(System.Char[])> method returns "123abc456xyz".
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The `TrimEnd(Char[])` method removes from the current string all trailing characters that are in the `trimChars` parameter. The trim operation stops when the first character that is not in `trimChars` is encountered at the end of the string. For example, if the current string is "123abc456xyz789" and `trimChars` contains the digits from "1" through "9", the `TrimEnd(Char[])` method returns "123abc456xyz".
```

File: xml/System/String.xml [hunk 16]
```diff
@@ -14149,26 +14356,26 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-   
-  
-## Examples  
- The following example demonstrates how you can use the <xref:System.String.TrimEnd%28System.Char%5B%5D%29> method to trim white space or punctuation marks from the end of a string.  
-  
- [!code-csharp[System.String.TrimEnd#2](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.String.TrimEnd/cs/sample2.cs#2)]
- [!code-vb[System.String.TrimEnd#2](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.String.TrimEnd/vb/Sample2.vb#2)]  
-  
- ]]></format>
+>  If the <xref:System.String.TrimEnd(System.Char[])> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing characters found in `trimChars` are removed from the current string.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
>  If the `TrimEnd(Char[])` method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all trailing characters found in `trimChars` are removed from the current string.
```

File: xml/System/String.xml [hunk 17]
```diff
@@ -14149,26 +14356,26 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
+## Examples
+
+The following example demonstrates how you can use the <xref:System.String.TrimEnd%2A?displayProperty=nameWithType> method to trim white space or punctuation marks from the end of a string.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The following example demonstrates how you can use the `TrimEnd(Char[])` method to trim white space or punctuation marks from the end of a string.
```

File: xml/System/String.xml [hunk 18]
```diff
@@ -14204,9 +14411,20 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+        <returns>The string that remains after all white-space characters are removed from the start of the current string. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimStart()> method removes from the current string all leading white-space characters. The trim operation stops when a non white-space character is encountered. For example, if the current string is "   abc   xyz    ", the <xref:System.String.TrimStart()> method returns "abc   xyz    ".
+
+> [!NOTE]
+>  If the <xref:System.String.TrimStart()> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading white space characters found in the current instance are removed.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
>  If the `TrimStart` method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading white space characters found in the current instance are removed.
```

File: xml/System/String.xml [hunk 19]
```diff
@@ -14236,10 +14454,21 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-        <remarks>To be added.</remarks>
+        <param name="trimChar">The Unicode character to remove.</param>
+        <summary>Removes all the leading occurrences of a specified character from the current string.</summary>
+        <returns>The string that remains after all occurrences of the <paramref name="trimChar" /> character are removed from the start of the current string. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimStart(System.Char)> method removes from the current string all leading `trimChar` characters. The trim operation stops when a character that is not `trimChar` is encountered. For example, if `trimChar` is `-` and the current string is "---abc---xyz----", the <xref:System.String.TrimStart(System.Char)> method returns "abc---xyz----".
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The `TrimStart(Char)` method removes from the current string all leading `trimChar` characters. The trim operation stops when a character that is not `trimChar` is encountered. For example, if `trimChar` is `-` and the current string is "---abc---xyz----", the `TrimStart(Char)` method returns "abc---xyz----".
```

File: xml/System/String.xml [hunk 20]
```diff
@@ -14236,10 +14454,21 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
+> [!NOTE]
+>  If the <xref:System.String.TrimStart(System.Char)> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading `trimChar` characters found in the current instance are removed.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
>  If the `TrimStart(Char)` method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading `trimChar` characters found in the current instance are removed.
```

File: xml/System/String.xml [hunk 21]
```diff
@@ -14290,36 +14519,36 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
       </Parameters>
       <Docs>
         <param name="trimChars">An array of Unicode characters to remove, or <see langword="null" />.</param>
-        <summary>Removes all leading occurrences of a set of characters specified in an array from the current <see cref="T:System.String" /> object.</summary>
-        <returns>The string that remains after all occurrences of characters in the <paramref name="trimChars" /> parameter are removed from the start of the current string. If <paramref name="trimChars" /> is <see langword="null" /> or an empty array, white-space characters are removed instead.</returns>
+        <summary>Removes all the leading occurrences of a set of characters specified in an array from the current string.</summary>
+        <returns>The string that remains after all occurrences of characters in the <paramref name="trimChars" /> parameter are removed from the start of the current string. If <paramref name="trimChars" /> is <see langword="null" /> or an empty array, white-space characters are removed instead. If no characters can be trimmed from the current instance, the method returns the current instance, unchanged.</returns>
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
        <returns>The string that remains after all occurrences of characters in the <paramref name="trimChars" /> parameter are removed from the start of the current string. If <paramref name="trimChars" /> is <see langword="null" /> or an empty array, white-space characters are removed instead. If no characters can be trimmed from the current instance, the method returns the current instance unchanged.</returns>
```

File: xml/System/String.xml [hunk 22]
```diff
@@ -14290,36 +14519,36 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-          <format type="text/markdown"><![CDATA[  
-  
-## Remarks  
- The <xref:System.String.TrimStart%2A> method removes from the current string all leading characters that are in the `trimChars` parameter. The trim operation stops when a character that is not in `trimChars` is encountered. For example, if the current string is "123abc456xyz789" and `trimChars` contains the digits from "1" through "9", the <xref:System.String.TrimStart%2A> method returns "abc456xyz789".  
-  
+          <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+The <xref:System.String.TrimStart(System.Char[])> method removes from the current string all leading characters that are in the `trimChars` parameter. The trim operation stops when a character that is not in `trimChars` is encountered. For example, if the current string is "123abc456xyz789" and `trimChars` contains the digits from "1" through "9", the <xref:System.String.TrimStart(System.Char[])> method returns "abc456xyz789".
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
The `TrimStart(Char[])` method removes from the current string all leading characters that are in the `trimChars` parameter. The trim operation stops when a character that is not in `trimChars` is encountered. For example, if the current string is "123abc456xyz789" and `trimChars` contains the digits from "1" through "9", the `TrimStart(Char[])` method returns "abc456xyz789".
```

File: xml/System/String.xml [hunk 23]
```diff
@@ -14290,36 +14519,36 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+
 > [!NOTE]
->  If the <xref:System.String.TrimStart%2A> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading white space characters found in the current instance are removed.  
-  
-   
-  
-## Examples  
- The following example demonstrates the basic functionality of the <xref:System.String.TrimStart%2A> method:
+>  If the <xref:System.String.TrimStart(System.Char[])> method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading characters that are in the `trimChars` parameter found in the current instance are removed.
```
Comment (rpetrusha) [inline xml/System/String.xml]:
```suggestion
>  If the `TrimStart(Char[])` method removes any characters from the current instance, this method does not modify the value of the current instance. Instead, it returns a new string in which all leading characters that are in the `trimChars` parameter found in the current instance are removed.
```

File: xml/System/String.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System/String.xml]: You're correct. I double checked [in the code](https://github.com/dotnet/corefx/blob/ac99a1b7168bd32046a954c3f06012c0fa909bed/src/Common/src/CoreLib/System/String.Manipulation.cs). I removed the mention of separator.

[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System/String.xml]: I'll also reword the mention of empty.

File: xml/System/String.xml [hunk 24]
```diff
@@ -8104,11 +8117,17 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="separator">The character to use as a separator. <paramref name="separator" /> is included in the returned string only if <paramref name="value" /> has more than one element.</param>
+        <param name="value">An array of strings to concatenate.</param>
+        <summary>Concatenates an array of strings, using the specified separator between each member.</summary>
+        <returns>A string that consists of the elements of <paramref name="value" /> delimited by the <paramref name="separator" /> character.
+
+-or-
+
+<see cref="F:System.String.Empty" /> if <paramref name="value" /> has no elements or <paramref name="separator" /> and all the elements of <paramref name="value" /> are <see cref="F:System.String.Empty" />.</returns>
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System/String.xml]: Reworded.

File: xml/System/String.xml [hunk 25]
```diff
@@ -8364,13 +8383,24 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
+        <param name="value">An array of strings to concatenate.</param>
+        <param name="startIndex">The first item in <paramref name="value" /> to concatenate.</param>
+        <param name="count">The number of elements from <paramref name="value" /> to concatenate, starting with the element in the <paramref name="startIndex" /> position.</param>
+        <summary>Concatenates an array of strings, using the specified separator between each member, starting with the element in <paramref name="value" /> located at the <paramref name="startIndex" /> position, and concatenating up to <paramref name="count" /> elements.</summary>
+        <returns>A string that consists of the elements of <paramref name="value" /> delimited by the <paramref name="separator" /> character.
+
+-or-
+
+<see cref="F:System.String.Empty" /> if <paramref name="count" /> is zero, <paramref name="value" /> has no elements, or <paramref name="separator" /> and all the elements of <paramref name="value" /> are <see cref="F:System.String.Empty" />.</returns>
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System/String.xml]: Fixed.

File: xml/System/String.xml [hunk 26]
```diff
@@ -12855,9 +13023,14 @@ A string is empty if it  is explicitly assigned an empty string ("") or <xref:Sy
...
-        <summary>To be added.</summary>
-        <returns>To be added.</returns>
+        <param name="provider">The provider to use to format the value.
+
+-or-
+
+<see langword="null" /> to obtain the numeric format information from the current locale setting of the operating system.
+        </param>
+        <summary>Returns the value of the current instance in a specified format.</summary>
+        <returns>The value of the current instance in the specified format.</returns>
```
[the remark this answers is not in the document]
Comment (carlossanlop) [reply xml/System/String.xml]: I'll use the return text used in [IConvertible.ToString](https://docs.microsoft.com/en-us/dotnet/api/system.iconvertible.tostring?view=netcore-3.0). Sounds good?

**Comment by rpetrusha** [comment]
I'll merge this, @carlossanlop. There's a broken link, but I'll open a separate PR to fix it.

**Pull request merged**